### PR TITLE
Randomize administrative password for new server creation.

### LINKED ISSUE #164: randomize admin passwords
Right now the root password for servers is always hard-coded to `"testpassword"`, this should instead be random per server.

### PR DESCRIPTION
**Pull request by sam-falvo**
Fixes #164

### CODE REVIEW

File: mimic/test/test_nova.py [hunk 1]
```diff
@@ -131,6 +135,8 @@ def test_get_server(self):
                          self.server_id)
         self.assertEqual(
             get_server_response_body['server']['status'], 'ACTIVE')
+        admin_password = get_server_response_body['server'].get('adminPass', None)
```
Comment (cyli) [inline mimic/test/test_nova.py]: Nice test!  Definitely shouldn't be giving out admin passwords in getting info.

**Comment by cyli** [comment]
Two nitpicks:

1.  Can we add a test to make sure that two created servers do not have the same admin password?
2.  Can we add one or more tests for the randomized string generation function?

I think the helper tests reside in https://github.com/rackerlabs/mimic/blob/master/mimic/test/test_util.py.

Other than that, this PR looks pretty great.  :)

File: mimic/util/helper.py [hunk 2]
```diff
@@ -15,6 +15,25 @@
...
+
+    :param int length: How long the string must be.
+    :param str selectable: If left unspecified, the random character selection
+        will be taken from uppercase and lowercase letters, digits, and a few
+        punctuation marks.  Otherwise, the characters will be taken from the
+        string provided.
+    :returns: A string of length `length`.
+    """
+    selectable = selectable or (
+        "abcdefghijklmnopqrstuvwxyz"
```
Comment (cyli) [inline mimic/util/helper.py]: Nonblocking - this is totally fine, and in fact may be more clear in some ways, but you can also use `strings.ascii_lowercase + strings.ascii_uppercase` (or just `strings.ascii_letters`), plus `strings.digits` :)

I'm not entirely sure about the symbols - I see that there exists a `strings.punctuation`, but not sure what that means - probably listing all the symbols is a better idea.

Anyway, totally nonblocking.  :)

File: mimic/test/test_util.py [hunk 3]
```diff
@@ -83,3 +83,27 @@ def test_unicode_body(self):
...
+    def test_selectable(self):
+        """
+        When passing a custom selectable, the results should derive only from
+        the characters you provide.
+        """
+        desired_chars = "02468"
+        for iteration in xrange(100):
+            a_string = helper.random_string(1024, selectable=desired_chars)
+            for char in a_string:
+                self.assertFalse(char not in desired_chars)
```
Comment (cyli) [inline mimic/test/test_util.py]: Non-blocking style nitpick:  the double negative may be a little harder to read than just a positive assertion :)

**Comment by cyli** [comment]
Up to you what you want to do about the non-blocking nitpicks :)  Otherwise LGTM if travis approves.

File: mimic/util/helper.py (hunk 2 above)
Comment (sam-falvo) [inline mimic/util/helper.py]: Cool, wasn't aware of this.

**Comment by sam-falvo** [comment]
I take your advice seriously, even the non-blockers.  Thanks!

**Comment by cyli** [comment]
Thanks!  Again, LGTM if travis +1's :)

**Pull request merged**